lib/twilio-ruby/framework/request.rb

```diff
@@ -0,0 +1,39 @@
+module Twilio
+  class Request
+    attr_reader :host, :port, :method, :url, :params, :data, :headers, :auth, :timeout
+
+    def initialize(host, port, method, url, params, data, headers, auth, timeout)
```

Comment:
We should probably make some of these optional and have sane defaults, for ex host, port, params, data, headers, timeout.